components/script/dom/bindings/str.rs

```diff
@@ -113,6 +113,74 @@ pub fn is_token(s: &[u8]) -> bool {
     })
 }
 
+/// Returns whether the language is matched, as defined by
+/// [RFC 4647](https://tools.ietf.org/html/rfc4647#section-3.3.2).
+pub fn extended_filtering(tag: &str, range: &str) -> bool {
+    let lang_ranges: Vec<&str> = range.split(',').collect();
+    let mut is_match: bool = true;
+
+    for lang_range in lang_ranges {
+        // step 1
+        let css_tags: Vec<&str> = lang_range.split('\x2d').collect();
+        let elem_tags: Vec<&str> = tag.split('\x2d').collect();
```

Comment:
Let's not call these css and elem tags. RFC 4647 does not define these terms. Instead, name them `range_subtags` and `tag_subtags`.